### PR TITLE
[Merged by Bors] - chore: Move lemmas about `Int.natAbs` and `zpowersHom`

### PR DESCRIPTION
**Pull request by YaelDillies**
These can be defined earlier for free.

Part of #9411

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

File: Mathlib/Data/Int/Order/Basic.lean [hunk 1]
```diff
@@ -550,5 +578,38 @@ theorem toNat_sub_of_le {a b : ℤ} (h : b ≤ a) : (toNat (a - b) : ℤ) = a -
 
 end Int
 
+section bit0_bit1
+variable {R}
+set_option linter.deprecated false
+
+-- The next four lemmas allow us to replace multiplication by a numeral with a `zsmul` expression.
+-- They are used by the `noncomm_ring` tactic, to normalise expressions before passing to `abel`.
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Int/Order/Basic.lean:586]: I know you're just moving this, but no they don't in lean 4. Maybe just drop the comment?

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**